samples/azure-function/Startup.cs

```diff
@@ -0,0 +1,31 @@
+using GSF.CarbonAware.Configuration;
+using Microsoft.Azure.Functions.Extensions.DependencyInjection;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using System.IO;
+
+[assembly: FunctionsStartup(typeof(CarbonAwareFunctions.Startup))]
+
+namespace CarbonAwareFunctions
+{
+    public class Startup : FunctionsStartup
+    {
+        public override void ConfigureAppConfiguration(IFunctionsConfigurationBuilder builder)
+        {
+            FunctionsHostBuilderContext context = builder.GetContext();
+
+            builder.ConfigurationBuilder
+                .AddJsonFile(Path.Combine(context.ApplicationRootPath, "appsettings.json"), optional: true, reloadOnChange: false)
+                .AddJsonFile(Path.Combine(context.ApplicationRootPath, $"appsettings.{context.EnvironmentName}.json"), optional: true, reloadOnChange: false)
+                .AddEnvironmentVariables();
+        }
+
+        public override void Configure(IFunctionsHostBuilder builder)
+        {
+            var configuration = builder.GetContext().Configuration;
+            builder.Services
+            .AddEmissionsServices(configuration)
+            .AddForecastServices(configuration);
+        }
```

Comment:
The general approach for Azure Function configuration is using environment variables and not the .NET specific ```appsettings.json```.

Configuration variables configured though a Json file like ```appsettings.json``` cannot be changed or overwritten in the [function app settings] (https://learn.microsoft.com/en-us/azure/azure-functions/functions-how-to-use-azure-function-app-settings?tabs=portal#settings) either.

Reply:
@juzuluag can correct me if I'm wrong but I believe this stisl allows for the function to be configured by environment variables. If you don't have the variables in app settings but have them in env vars, it would be loaded.

Reply:
It will work. My comment was around, all the Azure Functions examples uses [local.settings.json](https://learn.microsoft.com/en-us/azure/azure-functions/functions-develop-local#local-settings-file) file. Perhaps just make a note of it